src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodToStateMachineRewriter.cs

```diff
@@ -564,7 +564,7 @@ public override BoundNode VisitReturnStatement(BoundReturnStatement node)
                     F.Goto(_exprReturnLabel));
             }
 
-            return F.Goto(_exprReturnLabel);
+            return F.SequencePoint(null, F.Goto(_exprReturnLabel));
```

Comment:
Is the problem around implicit return statement? Otherwise we might need to do something similar in the body of the ```if``` above. Also, is adding hidden sequence point the right thing to do for an explicit return statements? At this point, I am thinking that the sequence point should be added when we add the implicit return instead.